lib/double-fetch-detection.c

```diff
@@ -0,0 +1,34 @@
+#include "linux/double-fetch-detection.h"
+#include <linux/context_tracking.h>
+#include <linux/entry-common.h>
+#include <linux/livepatch.h>
+#include <linux/audit.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+void addAddress(void* addr, unsigned long leng){
+           if(current->noRead==100 || current->noRead==-1)
```

Comment:
What do the constants mean? They should probably be named constants defined somewhere else.

Reply:
i think after reallocation if i need more memory i can delete constant"100".
also i can replace the checking of negative one to check on the pointer if it still not pointing to any address that means no system call need userspace address yet.